src/main/java/org/candlepin/subscriptions/tally/AccountUsageCalculationCache.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.tally;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.json.Event;
+import org.candlepin.subscriptions.util.DateRange;
+
+@Slf4j
+public class AccountUsageCalculationCache {
```

Comment:
Can you add a javadoc to describe that is the intention of this class?
Also, as far as I understand, this is acting like an accumulator of calculations instead of a cache, but I might be mistaken.

Reply:
Javadoc will be updated in next patch. It is used to cache calculations that are initially populated from snapshots in the DB when required. The intent is to track the state of the calculations over the entirety of the tally run so that we do not have to send Tally Summaries for every batch of Events that gets processed. This would result in a ton of summary messages per tally depending on the tally event batch size.

Reply:
DONE